### PR TITLE
New article: "Using pyenv"

### PR DESCRIPTION
**Pull request by dino-vinkovic**
Changes:
- Added new article "Using pyenv"
- https://beta.infinum.com/handbook/qa/automation/python/using-pyenv
- Added note on pyenv in Virtual Environments article

### CODE REVIEW

File: Automation/Python/Using pyenv.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    ```
+
+2. Install pyenv using [pyenv-installer](https://github.com/pyenv/pyenv-installer)
+    * This should install `pyenv` and a few useful plugins like `pyenv-virtualenv`
+
+    ```sh
+    ➜  ~ curl https://pyenv.run | zsh
+    ```
+
+3. Update PATH in `/.zprofile`. Add the following:
```
Comment (igor-pekovic) [inline Automation/Python/Using pyenv.md:42]: I don't have `zprofile`, instead I updated `.zshrc` file. Perhaps we can put a note about that in case there are similar differences.

**Review by igor-pekovic — APPROVED** [review]
I went through the process described and I could follow-up along with all the instructions, so I would say it's all good, at least for my setup 🙂 

I only have two comments regarding some minor improvements.

File: Automation/Python/Using pyenv.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+
+## Before you continue with pyenv installation
+
+Make sure [Homebrew](https://brew.sh/) is installed in the correct location. 
+If you have migrated from Mac using Intel architecture to one with ARM, you might have Homebrew still installed in the old location.
+
+Run: `which brew`
+
+Expected output: `/opt/homebrew`
```
[the remark this answers is not in the document]
Comment (dino-vinkovic) [reply Automation/Python/Using pyenv.md]: Would be more precise, I corrected it to what the output really says. 👍

File: Automation/Python/Using pyenv.md (hunk 1 above)
Comment (dino-vinkovic) [reply Automation/Python/Using pyenv.md:43]: All of those configuration files have to be created, they do not exist by default.

You can change `zshrc`, and many will use that one.
However, each of those configuration files has slightly different purpose and execution behaviour. So some might be better suited for certain actions.

Here are a few useful resources:
- https://unix.stackexchange.com/questions/71253/what-should-shouldnt-go-in-zshenv-zshrc-zlogin-zprofile-zlogout
- https://mac.install.guide/terminal/zshrc-zprofile

TLDR;
_Current recommended practice:

Use `~/.zprofile` to set the PATH and EDITOR environment variables.
Use `~/.zshrc` for aliases and a custom prompt, tweaking the appearance and behavior of the terminal._

Comment (dino-vinkovic) [reply Automation/Python/Using pyenv.md:43]: I will check other articles for info on configuration files and update it there if needed. I don't want to explain zsh in this one

Comment (igor-pekovic) [reply Automation/Python/Using pyenv.md:43]: I don't think you need to explain it, I just suggested you can add a note that some users might not have `.zprofile` file and might need to update file where `PATH` variable is defined.

Comment (dino-vinkovic) [reply Automation/Python/Using pyenv.md:43]: I get what you mean.

However, zprofile it the config file that is recommended to contain PATH. And it doesn't matter if you have some paths defined in other files. It will work.

But even if you do, because of that recommendation, we should not suggest files other than zprofile. Rather, people should update their config files.

That is why I think this topic requires a bit more info, and why I would stick to just saying "Use zprofile"

**Comment by endriina** [comment]
@dino-vinkovic 
_Automation/Python/Using pyenv.md_ and _Automation/Python/Virtual Environment.md_ loo good. 
Why is _Testing/Testing - Exploratory Testing.md_ deleted?

**Comment by dino-vinkovic** [comment]
@endriina good catch 🙏 
Deleted by mistake. It's back.
